types/types.go

```diff
@@ -412,12 +412,11 @@ func (m Member) String() string {
 
 // Signature is a function's signature.
 type Signature struct {
-	// TODO: store the parameter names, not just types.
-
 	// If a method of some type, this is the type it's a member of.
-	Receiver   *Type
-	Parameters []*Type
-	Results    []*Type
+	Receiver           *Type
+	Parameters         []*Type
+	ParametersWithName []Parameter
```

Comment:
`NamedParameters`

What do you think about instead just doing `ParameterNames []string`? It is awkward (I hate parallel arrays), but doesn't duplicate information or leave people scratching their heads quite as much about why we made the interface what it is.

Reply:
We can claim they're separate since if the parameters are unnamed, they must all be unnamed :)

Reply:
Yes, It makes sense. Using `ParameterNames []string` isn't the best practice, but it makes the interface clear.